src/basic/string-util.h

```diff
@@ -157,6 +157,8 @@ bool string_has_cc(const char *p, const char *ok) _pure_;
 
 char *ellipsize_mem(const char *s, size_t old_length_bytes, size_t new_length_columns, unsigned percent);
 char *ellipsize(const char *s, size_t length, unsigned percent);
+char *cellescape_impl(char *buf, size_t len, const char *s);
+#define cellescape(buf, s) cellescape_impl(buf, ELEMENTSOF(buf), s)
```

Comment:
uh, sorry for not noticing this eralier, but this one is too magic I think. I mean, it might be completely OK to allocate some buffer with new() or so and then pass it to cellescape, and this would fail horribly in that case... I'd avoid such magic.

Reply:
i still really don't like this bit tbh

Reply:
> it might be completely OK to allocate some buffer with new() or so and then pass it to cellescape

This will fail at compilation time, because ELEMENTSOF is smart enough to catch this. But OK, I'll rework this to use a define for the length and remove the macro.